### ISSUE TITLE
Sorting declarations

### ISSUE THREAD

**Issue by calda** [nicklockwood/SwiftFormat#1059]
I'm thinking about working on a way to sort declarations. One powerful approach for this could be to use `// swiftformat` comments to enable this on a per-type or per-range basis. Like:

```swift
enum MyEnum {
    // swiftformat:sort:begin
    case bar
    case baaz
    case foo
    // swiftformat:sort:end

    func unsortedFunction() { ... }
    func alsoUnsorted() { ... }
}
```

and:

```swift
// swiftformat:sort
class MyType {

    var bar: Bar {
        Bar(...)
    }

    func baaz() -> Baaz {
        Baaz(...)
    }

    var foo: Foo {
        Foo(...)
    }

}
```

`// swiftformat:sort` would also work with the `organizeDeclarations` rule to sort declarations within individual categories (but not break the category ordering). So if you're using `organizeDeclarations` then the `MyEnum` example would be sorted like:

```swift
// swiftformat:sort
enum MyEnum {
    case bar
    case baaz
    case foo

    func aSortedFunction() { ... }
    func sortedFunction() { ... }
    func thisFunctionIsAlsoSorted() { ... }
}
```

`parseDeclarations()` has already done a lot of the legwork to make this possible, so grabbing the list of declarations and then sorting them by name should be reasonably straightforward.

This is a bit of a different rule, since it's opt-in via comments. I was thinking it could be implemented as something like `FormatRules.sortDeclarations` and be enabled by default (but of course only actually have an effect when it sees the sort directive comments).

Any thoughts / interest?

**Comment by calda** [comment]
1068

**Comment by ZevEisenberg** [comment]
I think this can be closed since #1068 was merged?
